cardinal/world.go

```diff
@@ -0,0 +1,168 @@
+}
+
+// CreateMany creates multiple entities in the world, and returns the slice of ids for the newly created
+// entities. At least 1 component must be provided.
+func (w *World) CreateMany(num int, components ...AnyComponentType) ([]EntityID, error) {
+	return w.impl.CreateMany(num, toIComponentType(components)...)
+}
+
+// Create creates a single entity in the world, and returns the id of the newly created entity.
+// At least 1 component must be provided.
+func (w *World) Create(components ...AnyComponentType) (EntityID, error) {
+	return w.impl.Create(toIComponentType(components)...)
+}
+
+// Remove removes the given entity id from the world.
+func (w *World) Remove(id EntityID) error {
+	return w.impl.Remove(id)
+}
+
+// StartGame starts running the world game loop. After loopInterval time passes, a world tick is attempted.
+// In addition, an HTTP server (listening on the given port) is created so that game transactions can be sent
+// to this world. After StartGame is called, RegisterComponents, RegisterTransactions, RegisterReads, and AddSystem(s) may
+// not be called. If StartGame doesn't encounter any errors, it will block on the given done channel.
+func (w *World) StartGame(loopInterval time.Duration, disableSigVerification bool, port string, done chan struct{}) error {
+	if err := w.impl.LoadGameState(); err != nil {
+		return err
+	}
+	var opts []server.Option
+	if disableSigVerification {
+		opts = append(opts, server.DisableSignatureVerification())
+	}
+	opts = append(opts, server.WithPort(port))
+	txh, err := server.NewHandler(w.impl, opts...)
+	if err != nil {
+		return err
+	}
+	w.impl.StartGameLoop(context.Background(), loopInterval)
+	go txh.Serve()
+	<-done
+	return nil
+}
+
+// RegisterComponents adds the given components to the game world. After components are added, entities
+// with these components may be created. This function must only be called once.
+func (w *World) RegisterComponents(components ...AnyComponentType) error {
+	return w.impl.RegisterComponents(toIComponentType(components)...)
+}
+
+// RegisterTransactions adds the given transactions to the game world. HTTP endpoints to queue up/execute these
+// transaction will automatically be created when StartGame is called.
+func (w *World) RegisterTransactions(txs ...AnyTransaction) error {
+	return w.impl.RegisterTransactions(toITransactionType(txs)...)
+}
+
+// RegisterReads adds the given read capabilities to the game world. HTTP endpoints to use these reads
+// will automatically be created when StartGame is called.
+func (w *World) RegisterReads(reads ...AnyReadType) error {
+	return w.impl.RegisterReads(toIReadType(reads)...)
+}
+
+type WorldOption = ecs.Option
+
+func WithAdapter(adapter shard.Adapter) WorldOption {
+	return ecs.WithAdapter(adapter)
+}
+
+func WithReceiptHistorySize(size int) WorldOption {
+	return ecs.WithReceiptHistorySize(size)
+}
+
+func WithNamespace(namespace string) WorldOption {
+	return ecs.WithNamespace(namespace)
+}
```

Comment:
i know the underlying opts don't have comments, but maybe in this PR we could add some here.

Reply:
Yes, I'll add comments. query.go needs some more comments at well.